packages/dashboard/src/components/widgets/widget.tsx

```diff
@@ -34,14 +36,41 @@ const WidgetComponent: React.FC<WidgetProps> = ({
   const { x, y, z, width, height } = widget;
 
   // TODO: Replace with Redux dispatch
-  const [assets, setAssets] = useState<null | AssetQuery>(null);
+  const [assets, setAssets] = useState<null | AssetQuery[]>(null);
   const [internalWidget, setInternalWidget] = useState(widget);
+  const [error, setError] = useState<string | null>(null);
+  const isWritable = useWritableSiteWiseProperty();
+  const isInputWidget = internalWidget.componentTag === 'input';
 
   const [, drop] = useDrop(
     () => ({
       accept: ItemTypes.ResourceExplorerAssetProperty,
-      drop: ({ queryAssetsParam }: { queryAssetsParam: AssetQuery }) => {
-        setAssets(queryAssetsParam);
+      drop: async ({ queryAssetsParam }: { queryAssetsParam: AssetQuery[] }) => {
+        // TODO: decouple this from drop callback
+        if (isInputWidget) {
+          const assetId = queryAssetsParam[0]?.assetId;
+          const propertyId = queryAssetsParam[0]?.properties?.[0].propertyId;
+          const { writable } = await isWritable({ assetId, propertyId });
+          if (writable) {
+            setAssets(queryAssetsParam);
+          }
+        } else {
+          setAssets(queryAssetsParam);
+        }
+        setError(null);
```

Comment:
Blocking: Abstract into symbol. It's not clear what is happening here at first glace.

Reply:
Could you help me understand what you mean by abstract into symbol? This will just set the error state to null which means we won't show an error message.